### PR TITLE
Edit-InstallmentPage

### PR DESCRIPTION
**Pull request by AsmaMohame**

### CODE REVIEW

File: retail-managment-system/rms-ui/src/assets/i18n/ar.json [hunk 1]
```diff
@@ -135,11 +135,13 @@
     "due-date": "تاريخ الاستحقاق",
     "payment-date": "تاريخ الدفع",
     "status": "الحاله",
+    "customerName":" اسم العميل",
```
Comment (AbdelrahmanMohyeldin) [inline retail-managment-system/rms-ui/src/assets/i18n/ar.json:139]: you can use dash case like other names

Comment (AsmaMohame) [reply retail-managment-system/rms-ui/src/assets/i18n/ar.json]: done

**Review by AbdelrahmanMohyeldin — APPROVED** [review]

File: retail-managment-system/rms-ui/src/app/pages/installments/edit-installment/edit-installment.component.html [hunk 2]
```diff
@@ -0,0 +1,44 @@
+
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/installments/edit-installment/edit-installment.component.html:1]: "edit-installment" or "pay-installment" better for component name?

File: retail-managment-system/rms-ui/src/app/pages/installments/edit-installment/edit-installment.component.scss [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    color: red;
+}
+.btn {
+    background-color: #002d40;
+    width: 80px;
+    height: 40px;
+  }
+
+  .print {
+    display: none;
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/installments/edit-installment/edit-installment.component.scss:18]: you used this also as bootstrap. which work?

File: retail-managment-system/rms-ui/src/app/pages/installments/edit-installment/edit-installment.component.scss [hunk 4]
```diff
@@ -0,0 +1,24 @@
+.content {
+    width: 400px;
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/installments/edit-installment/edit-installment.component.scss:2]: why used px not %

File: retail-managment-system/rms-ui/src/app/pages/installments/edit-installment/edit-installment.component.ts [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+})
+export class EditInstallmentComponent implements OnInit {
+  installForm!: FormGroup;
+  isButtonVisible!: boolean;
+  submit: boolean = false;
+  paymentAmount!: number;
+  paymentDate!: Date;
+ constructor(private installmentrepositry: InstallmentRepositry,
+  private formBuilder : FormBuilder,
+  private _snackBar: MatSnackBar,
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/installments/edit-installment/edit-installment.component.ts:26]: use camel case

File: retail-managment-system/rms-ui/src/app/pages/installments/edit-installment/edit-installment.component.ts [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+    {
+     this.installForm.controls['status'].setValue(1) ;
+    }
+    else{
+      this.installForm.controls['status'].setValue(0) ;
+    }
+    
+  }
+
+ clickButton(){
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/installments/edit-installment/edit-installment.component.ts:105]: use better name convention for function name

File: retail-managment-system/rms-ui/src/app/pages/installments/edit-installment/edit-installment.component.ts [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+    this.paymentDate=this.installForm.controls['paymentDate'].value
+   }
+  }
+
+  updateInstallment() {
+    this.isButtonVisible = true;
+    this.submit = true;
+    this.installmentrepositry.update(this.installForm.value).subscribe(() => {
+     this.clickButton();
+      this.dialog.closeAll();
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/installments/edit-installment/edit-installment.component.ts:64]: use a function that you already added for close
to achieve single responsibility

File: retail-managment-system/rms-ui/src/app/pages/installments/edit-installment/edit-installment.component.ts [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+    })
+  }
+
+  onSubmit()  {
+    this.installForm.markAllAsTouched();
+    if (this.installForm.valid) {
+      this.installForm.controls['id'].setValue(this.editData.id) ;
+      this.installForm.controls['paymentDate'].setValue((new Date()).toISOString().substring(0,10))
+      this.paymentAmount = this.installForm.controls['paymentAmount'].value
+      this.paymentDate = this.installForm.controls['paymentDate'].value ;
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/installments/edit-installment/edit-installment.component.ts:81]: line 80 and 81 is duplicated in fetchData function why?

line 78 can add this in fetchData and delete from here 
to make data that will be fetch in same place

File: retail-managment-system/rms-ui/src/app/pages/installments/edit-installment/receipt/receipt.component.html [hunk 9]
```diff
@@ -0,0 +1,42 @@
+<div class="panel-heading   mb-1 mt-5 mb-5 w-100 text-center" >
+  <h3 class="panel-title w-100 text-cprint-sectionenter fw-bold ">
+    Installment Receipt
+  </h3>
+</div>
+<div class="main d-flex justify-content-center">
+    <table class="table">
+      <tbody>
+        <tr>
+          <th scope="col">Name :</th>
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/installments/edit-installment/receipt/receipt.component.html:10]: use translation

File: retail-managment-system/rms-ui/src/app/pages/installments/installment.component copy.html [hunk 10]
```diff
@@ -0,0 +1,119 @@
+<div class="main">
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/installments/installment.component copy.html:1]: what is this file?

File: retail-managment-system/rms-ui/src/app/pages/installments/edit-installment/receipt/receipt.component.ts [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+import { MatDialog, MAT_DIALOG_DATA } from '@angular/material/dialog';
+import { Installment } from 'src/app/domain/installment/models/installment';
+
+@Component({
+  selector: 'app-receipt',
+  templateUrl: './receipt.component.html',
+  styleUrls: ['./receipt.component.scss']
+})
+export class ReceiptComponent  {
+  print: boolean =false
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/installments/edit-installment/receipt/receipt.component.ts:11]: are you need this property?

File: retail-managment-system/rms-ui/src/app/pages/installments/edit-installment/receipt/receipt.component.ts [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+ @Input()install!: Installment;
+ @Input() payAmount!: number;
+ @Input() payDate!: Date;
+
+  constructor(@Inject (MAT_DIALOG_DATA) public editData : Installment,  private dialog: MatDialog,) { }
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/installments/edit-installment/receipt/receipt.component.ts:16]: are you use this injects ?

File: retail-managment-system/rms-ui/src/app/pages/installments/installment.component.ts [hunk 13]
```diff
@@ -0,0 +1,105 @@
...
+    applyFilter(filterValue: string) {
+      const tableFilters :any []= [];
+      tableFilters.push({
+        id:'',
+        value: filterValue
+      });
+      this.dataSource.filter = JSON.stringify(tableFilters);
+    }
+
+  setDate(install : Installment)
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/installments/installment.component.ts:88]: use better name convention(meaningful) for function name

**Review by ZinabHashem — CHANGES_REQUESTED** [review]

**Comment by ZinabHashem** [comment]
image

**Pull request merged**